int64 currentDigit
int64 nextDigit
